### ISSUE TITLE
Prepare Odoo 15

### ISSUE THREAD

**Issue by sbidoul** [OCA/oca-addons-repo-template#69]
Upgrade linters, while avoiding unnecessary disruption which make fw/backports difficult.

**Comment by sbidoul** [comment]
@moylop260 et al, can you look at https://github.com/OCA/oca-addons-repo-template/blob/master/src/.pylintrc.jinja and https://github.com/OCA/oca-addons-repo-template/blob/master/src/.pylintrc-mandatory.jinja and suggest new checks to add or move for v15 ?

**Comment by moylop260** [comment]
Hi @sbidoul thanks for you efforts on this

I just got the diff between the pylint-odoo checks enabled in the configuration file vs all the current ones in pylint-odoo

The diff are the following checks:
  - attribute-string-redundant
  - character-not-valid-in-resource-link
  - consider-merging-classes-inherited
  - context-overridden
  - copy-wo-api-one
     - It is not valid for new versions
  - create-user-wo-reset-password
  - dangerous-filter-wo-user
  - dangerous-qweb-replace-wo-priority
  - deprecated-data-xml-node
  - deprecated-openerp-xml-node
  - duplicate-po-message-definition
  - except-pass
  - file-not-used
  - invalid-commit
  - javascript-lint
    - It is supported by pre-commit eslint
  - manifest-maintainers-list
  - missing-import-error
    - It is not needed anymore
  - missing-manifest-dependency
    - It is not updated to new way to use pypi packages name directly
  - missing-newline-extrafiles
  - missing-readme
  - missing-return
  - no-utf8-coding-comment
    - It is not needed anymore
  - odoo-addons-relative-import
  - old-api7-method-defined
  - po-msgstr-variables
  - po-syntax-error
  - renamed-field-parameter
  - resource-not-exist
  - str-format-used
  - test-folder-imported
  - translation-contains-variable
  - translation-positional-used
  - unnecessary-utf8-coding-comment
  - website-manifest-key-not-valid-uri
  - xml-attribute-translatable
  - xml-deprecated-qweb-directive
  - xml-deprecated-tree-attribute

**Comment by pedrobaeza** [comment]
cc @Yajo

**Comment by Yajo** [comment]
> Upgrade linters, while avoiding unnecessary disruption which make fw/backports difficult.

I wonder if we should start using [darker](https://pypi.org/project/darker/) instead of black, to help on this. What do you think?

**Comment by pedrobaeza** [comment]
I don't think we should add another tool more to the chain, and it's not so critical to always pass pre commit in a separate previous commit.

**Comment by moylop260** [comment]
There are a few of out-of-the-box pylint checks that could be good to have

 - Use a generator instead 'any(field in vals for field in m2m_fields)' for `any([field in vals for field in m2m_fields])`
 - logging-not-lazy Specify string format args as loggin funct parameters.
    - Use `_logger.warning('Msg with a param %s' % param1)` instead of `_logger.warning('Msg with a param %s', param1)`
    - A bad parsing won't raise an error since that it only is a logger not logic and the string won't be evaluated early
 - consider-using-with opening files `with open() as f: f.read()` instead of `f=open();f.read();f.close()`

**Comment by sbidoul** [comment]
> There are a few of out-of-the-box pylint checks that could be good to have

+1 to add more on 15.0. But not on previous branches to avoid making existing branches red.

**Comment by Yajo** [comment]
All these checks you say @moylop260 should go to mandatory or optional checks?

**Comment by sbidoul** [comment]
I'm personally happy with adding mandatory checks in v15. We can till move them to optional if they cause problems.

**Comment by moylop260** [comment]
@Yajo @sbidoul 

FYI I'm using a configuration file to enable all the pylint checks then disable what we don't like or not valid for Odoo

This is my list:
 - https://github.com/Vauxoo/pylint-conf/blob/043d46066a62a8cb0db12ebb680c105a24397777/conf/pylint_vauxoo_light.cfg#L21-L82

I just share the information no action required here
